assets/license-headers-check

```diff
@@ -1,15 +1,33 @@
 #!/usr/bin/env bash
 #
-# Iterates over all rust source files and ensures they start with the license
-# header as per the `.license-header` file at the root of the repository.
+# Iterates over all source files and ensures they start with the license header.
+#
+# If run with the `--fix` option license headers are prepended if they are
+# missing.
 
 set -euo pipefail
-IFS=$'\n'
-
 shopt -s globstar extglob
-for file in */+(src|tests|examples)/**/*.rs
-do
-    rustfmt --config license_template_path=".license-header" --color never --quiet --check "$file" || {
-        sed -i -e '1r.license-header' -e '1{h;d}' -e '2{x;G}' "$file"
-    }
+
+license_header="\
+// <project> is Copyright © 2019-2020
+// [The Radicle Foundation <hello@radicle.foundation> OR Monadic GmbH <company@monadic.xyz>].
+//
+// This file is part of <project>,
+// distributed under the GPL v3 with Radicle Linking Exception.
+// For full terms see the included LICENSE file"
```

Comment:
I get that you want to document a script that works with non-Rust projects. It should, however, not assume hard-coded copyright years and owners, see #6

Reply:
Yes, I would like to avoid requirement of rust tools where they aren't relevant. Is rustfmt even capable of checking only the header and ignoring the non-rust body?

Do we need to have different development years and main contributors on different files? I honestly don't know, on one hand each file is a separate entity, but on the other they are all parts of a single codebase and usually don't make much sense alone.

Reply:
Do we need to support different development years and authors on each file? I honestly don't know, I couldn't find anything on the internet. On one hand each file is a separate entity, but on the other they are merely parts of one project under a single copyright.

Reply:
> they are merely parts of one project under a single copyright

No they aren't -- the copyright is held by the author. By contributing as a non-employee/contractor of Monadic / Radicle Foundation, you don't automatically transfer your rights to Monadic / Radicle Foundation -- you get to keep it. This means that, if you contribute a substantial work, you put your own name there.

The copyright year is supposed to be the **year of first publication**. 

This means:

* For license headers, one puts the **current** year in a new file. If you are an "internal" contributor, you put either Monadic or Foundation, otherwise you put your own name.
* We don't want to constantly update the copyright notice in the `LICENSE` file. So there, I propose in #6 to use "The <project> Contributors" as the copyright holders.

The question of whether or not the copyright year has to be updated can be answered as follows: the year is the year of first publication (of the work as a whole, or the individual file). It is the **claim** of copyright starting with this year -- 70 years from that year, the copyright will expire, and the work will fall into the public domain. So -- do we need to extend the range every year? I don't think so, once every 70 years should be sufficient.

IANAL, ofc.

Reply:
I've adjusted it for these rules